### PR TITLE
fix https, no need to check youtu.be, fix timestamping

### PR DESCRIPTION
**Pull request by MonMonMnemonics**

### CODE REVIEW

File: src/js/mchad.js [hunk 1]
```diff
@@ -60,9 +58,20 @@ export async function getArchiveFromRoom(room) {
       link: room.Link
     })
   }).then(toJson).catch(() => []);
-  const { Stime: firstTime } = [...script, { Stime: 0 }, { Stime: 0 }][1];
+  
+  var { Stime: firstTime } = [...script, { Stime: 0 }][0];
```
Comment (r2dev2) [inline src/js/mchad.js]: Please use ``const`` or ``let`` instead of ``var``.

File: src/js/mchad.js [hunk 2]
```diff
@@ -60,9 +58,20 @@ export async function getArchiveFromRoom(room) {
...
+  var startmatch = script.filter(e => (e.Stext.match(/--.*Stream.*Start.*--/i) != null));
+  if (startmatch.length != 0){
+    firstTime = startmatch[0].Stime;
+  }
+  /*
+  //  start also doesn't seem to be correct for some reason ?
+  if (firstTime != 0){
+    firstTime = start;
+  }
+  */
```
Comment (r2dev2) [inline src/js/mchad.js]: Please remove the commented out code

File: src/js/mchad.js [hunk 3]
```diff
@@ -75,6 +84,7 @@ export const getArchive = videoId => readable(null, async set => {
 
   const script = await getArchiveFromRoom(vod[0])
     .then(s => s.map(addUnix))
+    .then(s => s = s.filter(e => e.unix >= 0))
```
Comment (r2dev2) [inline src/js/mchad.js]: No need to reassign to ``s``.

**Review by r2dev2 — CHANGES_REQUESTED** [review]

File: src/js/mchad.js (hunk 1 above)
Comment (MonMonMnemonics) [reply src/js/mchad.js]: done

File: src/js/mchad.js (hunk 2 above)
Comment (MonMonMnemonics) [reply src/js/mchad.js]: done

File: src/js/mchad.js (hunk 3 above)
Comment (MonMonMnemonics) [reply src/js/mchad.js]: done

**Comment by MonMonMnemonics** [comment]
All requested changes are done, review please.

**Review by r2dev2 — APPROVED** [review]

**Pull request merged**